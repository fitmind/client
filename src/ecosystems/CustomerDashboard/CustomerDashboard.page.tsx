import React from 'react';
import { Jumbotron } from 'react-bootstrap';

const CustomerDashboard: React.FC = () => (
    <Jumbotron>
        <h1>Customer Dashboard</h1>
    </Jumbotron>
);

export default CustomerDashboard;
