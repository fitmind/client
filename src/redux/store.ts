import { Store, createStore, applyMiddleware } from 'redux';
import { ApplicationState, storeInitialState, createRootReducer } from './reducers/root.reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import { rootSaga } from './saga/root.saga';
import { History, createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

// We use hash history because this example is going to be hosted statically.
// Normally you would use browser history.
export const history: History = createBrowserHistory();

export const configureStore = (initialState: ApplicationState): Store<ApplicationState> => {
    const sagaMiddleware = createSagaMiddleware();
    const composeEnhancers = composeWithDevTools({});
    let store = createStore(
        createRootReducer(history),
        initialState,
        composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware)),
    );

    sagaMiddleware.run(rootSaga);
    return store;
};

const store: Store<ApplicationState> = configureStore(storeInitialState);

export default store;
