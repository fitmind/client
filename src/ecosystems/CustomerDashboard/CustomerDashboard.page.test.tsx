import React from 'react';
import { shallow } from 'enzyme';
import CustomerDashboard from './CustomerDashboard.page';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store, { history } from '../../redux/store';
import { Router } from 'react-router';

describe('Customer Dashboard page', () => {
    it('renders without crashing', () => {
        shallow(
            <Provider store={store}>
                <Router history={history}>
                    <CustomerDashboard />
                </Router>
            </Provider>,
        );
    });

    it('should match the snapshot', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router history={history}>
                    <CustomerDashboard />
                </Router>
            </Provider>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
