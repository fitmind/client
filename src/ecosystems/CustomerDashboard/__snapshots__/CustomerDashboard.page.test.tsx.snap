// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Dashboard page should match the snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    />
    <div
      className="col"
    >
      <div
        className="sc-bdVaJa hBYjom"
      >
        <div
          className="sc-bwzfXH dhFCHk"
        >
          Welcome 
          !
        </div>
        <h5>
          Here you will be able to look at your upcoming and past appointments. Message your coaches and get insights on your improvements!
        </h5>
      </div>
    </div>
    <div
      className="col-md-4"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-1"
    />
    <div
      className="col-md-10"
    >
      <div
        className="sc-htpNat fYCKZt"
      >
        <h3>
          Upcoming Appointments
        </h3>
        <div
          className="table-responsive-md"
        >
          <table
            className="table"
          >
            <thead>
              <tr>
                <th>
                  #
                </th>
                <th>
                  Client
                </th>
                <th>
                  Time
                </th>
                <th>
                  Email
                </th>
                <th>
                  Listing
                </th>
                <th>
                  Price
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
        <h3>
          Past Appointments
        </h3>
        <div
          className="table-responsive-md"
        >
          <table
            className="table"
          >
            <thead>
              <tr>
                <th>
                  #
                </th>
                <th>
                  Client
                </th>
                <th>
                  Time
                </th>
                <th>
                  Email
                </th>
                <th>
                  Listing
                </th>
                <th>
                  Price
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
    <div
      className="col-md-1"
    />
  </div>
</div>
`;
